Payment test failure

test_container_payments[EC 2/1 CBF 1-3-1]:

> AssertionError: invalid number of withdrawals from the user wallet per epoch
assert 0 == 1
> +  where 0 = len([])

https://rest.fs.neo.org/Aku38wpn2xEqfEPxmwyAa7N3VnAkMV8PEVkpLqwKM1z3/5289-1778075546/index.html#suites/44e11ced39071e8d0cfc11c5b94622ba/aaa63bdf935b3a0f/

--- a/pytest_tests/lib/neofs_env/neofs_epoch.py
+++ b/pytest_tests/lib/neofs_env/neofs_epoch.py
@@ -35,11 +35,19 @@ def wait_for_epochs_align(neofs_env: NeoFSEnv, epoch_number: Optional[int] = Non
 
 @allure.step("Wait until new epoch arrives")
 @wait_for_success(60, 1)
-def wait_until_new_epoch(neofs_env: NeoFSEnv, current_epoch: int) -> int:
-    for node in neofs_env.storage_nodes:
-        next_epoch = get_epoch(neofs_env, node)
-        assert next_epoch == current_epoch + 1, "Next epoch didn't arrive during timeout"
-    return next_epoch
+def wait_until_new_epoch(
+    neofs_env: NeoFSEnv,
+    current_epoch: int,
+    require_all_storage_nodes: bool = True,
+) -> int:
+    expected = current_epoch + 1
+    epochs = [get_epoch(neofs_env, node) for node in neofs_env.storage_nodes]
+    matched = sum(1 for epoch in epochs if epoch == expected)
+    required = len(epochs) if require_all_storage_nodes else 1
+    assert matched >= required, (
+        f"Next epoch didn't arrive during timeout (expected {expected} on {required} node(s), got {epochs})"
+    )
+    return expected
 
 
 @allure.step("Get Epoch")

--- a/pytest_tests/tests/payment/test_container_payments.py
+++ b/pytest_tests/tests/payment/test_container_payments.py
@@ -5,7 +5,7 @@ import allure
 import neofs_env.neofs_epoch as neofs_epoch
 import pytest
 from helpers.complex_object_actions import get_nodes_with_object
-from helpers.container import create_container
+from helpers.container import create_container, delete_container
 from helpers.file_helper import generate_file
 from helpers.neofs_verbs import put_object
 from helpers.wallet_helpers import create_wallet_with_money, get_neofs_balance
@@ -37,6 +37,7 @@ class TestContainerPayments:
     )
     def test_container_payments(
         self,
+        request: pytest.FixtureRequest,
         neofs_env_with_mainchain: NeoFSEnv,
         wallet_with_money: NodeWallet,
         container_rule: str,
@@ -61,6 +62,17 @@ class TestContainerPayments:
                 shell=neofs_env.shell,
                 endpoint=neofs_env.sn_rpc,
             )
+
+            def cleanup_container() -> None:
+                with allure.step("Delete container"):
+                    delete_container(
+                        wallet_with_money.path,
+                        cid,
+                        shell=neofs_env.shell,
+                        endpoint=neofs_env.sn_rpc,
+                    )
+
+            request.addfinalizer(cleanup_container)
 
             user_wallet_balance_after_container_creation = get_neofs_balance(
                 neofs_env, neofs_env.neofs_cli(wallet_with_money.cli_config), wallet_with_money

--- a/pytest_tests/tests/payment/test_container_payments.py
+++ b/pytest_tests/tests/payment/test_container_payments.py
@@ -105,7 +105,7 @@ class TestContainerPayments:
 
         with allure.step("Wait for a couple of epochs to arrive"):
             new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, neofs_epoch.get_epoch(neofs_env))
-            new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, new_epoch)
+            new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, new_epoch, require_all_storage_nodes=False)
 
         with allure.step("Ensure the user wallet balance is charged only once per epoch"):
             deltas = []
@@ -125,7 +125,7 @@ class TestContainerPayments:
 
         with allure.step("Wait for a couple of epochs to arrive"):
             new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, neofs_epoch.get_epoch(neofs_env))
-            new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, new_epoch)
+            new_epoch = neofs_epoch.wait_until_new_epoch(neofs_env, new_epoch, require_all_storage_nodes=False)
 
         with allure.step("Ensure the storage node balance is debited only once per epoch"):
             sn, _ = next(((sn, sn_info) for sn, sn_info in storage_nodes_info.items() if sn_info["objects_count"] > 0))

